### DISCUSSION TITLE
Markers on android

### DISCUSSION THREAD

**Discussion by ozan123456789** [mapsforge/mapsforge#1344] [Q&A] [answered]
I am trying to add two different markers with different images but both of them show the image of the last declared marker.

```
`Marker marker=createMarker("sdcard/Android/data/org.mapsforge.samples.android/files/a.png",getGps());
        Marker marker2=createMarker("sdcard/Android/data/org.mapsforge.samples.android/files/Yellowgemi.png",getGps());
        TileStoreLayer tileStoreLayer = new TileStoreLayer(tW,
                this.mapView.getModel().mapViewPosition, AndroidGraphicFactory.INSTANCE, false);
        //TileStoreLayer tileStoreLayerSea = new TileStoreLayer(tSea,
        //        this.mapView.getModel().mapViewPosition, AndroidGraphicFactory.INSTANCE, false);

        mapView.getLayerManager().getLayers().add(tileStoreLayer);
        mapView.getLayerManager().getLayers().add(marker);
        mapView.getLayerManager().getLayers().add(marker2);`

```
My purpose is to add more than 10 markers to the map and every marker should have different images.

**Comment by devemux86** [comment, marked as answer]
See the sample [OverlayMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java) (with [Utils](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/Utils.java)).
